src/Compilers/CSharp/Portable/Syntax/LambdaUtilities.cs

```diff
@@ -424,6 +419,13 @@ internal static bool IsClosureScope(SyntaxNode node)
                     return true;
 
                 default:
+                    // With the introduction of pattern-matching, many nodes now contain top-level
+                    // expressions that may introduce pattern variables.
+                    if (node.Parent.IsKind(SyntaxKind.EqualsValueClause))
```

Comment:
Consider moving out of the switch, since all the switch cases above return.
Also, I didn't understand why the check is now on `node.Parent` instead of `node` itself.

Reply:
> Consider moving out of the switch, since all the switch cases above return.

I don't believe it matters what the other cases do. 

> I didn't understand why the check is now on node.Parent instead of node itself.

See changes in RewriteFieldInitializer

---
In reply to: [174218563](https://github.com/dotnet/roslyn/pull/25441#discussion_r174218563) [](ancestors = 174218563)